src/atomate2/vasp/flows/defect.py

```diff
@@ -22,27 +33,160 @@
     use_structure_charge=True,
-    user_incar_settings=DEFECT_INCAR_SETTINGS,
-    user_kpoints_settings=DEFECT_KPOINT_SETTINGS,
+    user_incar_settings={
+        "LAECHG": False,
+        "LREAL": False,
+        "ALGO": "Normal",
+        "NSW": 99,
+        "LCHARG": False,
+        "HFSCREEN": 0.2,
+        "LHFCALC": True,
+        "PRECFOCK": "Fast",
+        "LASPH": True,
+        "LDAU": False,
+    },
+    user_kpoints_settings=SPECIAL_KPOINT,
+)
+
+HSE_DOUBLE_RELAX = DoubleRelaxMaker(
+    relax_maker1=RelaxMaker(input_set_generator=PBE_GEN),
+    relax_maker2=RelaxMaker(
+        input_set_generator=HSE_GEN,
+        task_document_kwargs={"store_volumetric_data": ["locpot"]},
+    ),
 )
-DEFECT_STATIC_GENERATOR = StaticSetGenerator(
-    user_incar_settings=DEFECT_INCAR_SETTINGS,
-    user_kpoints_settings=DEFECT_KPOINT_SETTINGS,
+
+DEFECT_STATIC_GENERATOR: StaticSetGenerator = StaticSetGenerator(
+    user_kpoints_settings=SPECIAL_KPOINT,
 )
 
+CCD_DEFAULT_DISTORTIONS = (-1, -0.15, -0.1, -0.05, 0, 0.05, 0.1, 0.15, 1)
+
+
+# Define RelaxMaker Errors
+def check_relax_maker(maker: Maker):
+    """Check specific RelaxMaker settings.
+
+    Check all nested RelaxMaker in to make sure specific Setting is set.
+
+    Parameters
+    ----------
+    maker : Maker
+        The Maker object to check.
+    """
+
+    def check_func(relax_maker: RelaxMaker):
+        input_gen = relax_maker.input_set_generator
+        vis = input_gen.get_input_set(DUMMY_STRUCT, potcar_spec=True)
+        if input_gen.use_structure_charge is False:
+            raise ValueError("use_structure_charge should be set to True")
+        if vis.incar["LVHAR"] is False:
+            raise ValueError("LVHAR should be set to True")
+        if vis.incar["ISIF"] != 2:
+            raise ValueError("ISIF should be 2")
+
+    # TODO: add this check after JobFlow is updated to support recursive_call
+    # recursive_call(maker, func=check_func, class_filter=RelaxMaker, nested=True)
+
+
+@dataclass
+class FormationEnergyMaker(Maker):
+    """Maker class to help calculate of the formation energy diagram.
+
+    Attributes
+    ----------
+    name: str
+        The name of the flow created by this maker.
+    relax_maker: .BaseVaspMaker or None
+        A maker to perform a atomic-position-only relaxation on the defect charge
+        states. If None, the defaults will be used.
+    validate_maker: bool
+        If True, the code will check the relax_maker for specific settings
+    """
+
+    name: str = "formation energy"
+    validate_maker: bool = True
+    relax_maker: BaseVaspMaker = field(
+        default_factory=lambda: RelaxMaker(
+            input_set_generator=PBE_GEN,
+            task_document_kwargs={"store_volumetric_data": ["locpot"]},
+        )
+    )
+
+    def __post_init__(self):
+        """Check the calculation settings."""
+        if self.validate_maker:
+            check_relax_maker(self.relax_maker)
```

Comment:
I think these checks should be removed. If people want to use wrong settings they should be able to. In my opinion we should remove this and just add a warning in the docstring to say that these options should be set on any input makers. We don't currently check that other settings work for other workflows and if we start going down that route then it will result in quite convoluted code. Especially when you can have nested flows it starts getting very difficult to check properly and the checks could be quite brittle. For example, what happens if you want to do an initial quick relaxation without LVHAR, and turn on the required settings afterwards. This will currently result in an error.